stream-chat-android-client/src/main/java/io/getstream/chat/android/client/StreamLifecycleObserver.kt

```diff
@@ -22,54 +22,49 @@ import androidx.lifecycle.LifecycleOwner
 import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
 import io.getstream.logging.StreamLog
 import kotlinx.coroutines.withContext
+import java.util.concurrent.atomic.AtomicBoolean
 
-internal class StreamLifecycleObserver(
-    private val lifecycle: Lifecycle,
-    private val handler: LifecycleHandler,
```

Comment:
Instead of injecting the handler by the constructor, we add/remove it when it is needed.
With this new approach, this Observer can be reused across our SDK and we don't need to have multiple instances handling the Android Livecycle.